### PR TITLE
Feat delete empty folders

### LINKED ISSUE #47: Wrongly removes conan metadata files
Thank you for your awesome tool!

We would like to use it for cleaning up a conan repository.

When the conan client deploys a package, it uploads:
* conan_package.tgz

At the same time, the server side generates some metadata (see also https://github.com/conan-io/conan/issues/8418)
* .timestamp
* index.json

When dowloading a conan package, it only downloads package.tar.gz, but not the additional meta files.

If I now run the `rules.delete_not_used_since(days=60),` rule on packages which were deployed before 60 days, but still regularly downloaded, it will delete the meta files, but keep the conan_package.tgz file. A conan package download does not trigger the download counter on the metadata files.

And this will invalidate the whole conan package (as also mentioned here https://github.com/jfrog/artifactory-user-plugins/issues/319).

One solution would be to use `rules.exclude_filename(['.timestamp', 'index.json'])` (also suggested here https://github.com/jfrog/artifactory-user-plugins/pull/399).
This would then only delete the package, but not the metadata. And therefore, the metadata is never cleaned up... So also not the perfect solution.

Do you have any suggestions how to handle this?
I guess to solve this we would need dedicated rules just for conan, similar to the docker rules?

There may also other package types affected by this (e.g., npm, pypi, conda, debian, ...), see also https://github.com/jfrog/artifactory-user-plugins/issues/319#issuecomment-964064737

### PR DESCRIPTION
**Pull request by allburov**
Fix #47

### CODE REVIEW

**Comment by allburov** [comment]
@Pro could you test these changes and look at the code please?

**Comment by allburov** [comment]
@Pro did you have a change to test these changes?

File: artifactory_cleanup/rules/base.py [hunk 1]
```diff
@@ -200,7 +200,12 @@ def filter(self, artifacts):
         return artifacts
 
     def delete(self, artifact, destroy):
-        artifact_path = quote("{repo}/{path}/{name}".format(**artifact))
+        if artifact["path"] == ".":
+            artifact_path = "{repo}/{name}".format(**artifact)
+        else:
+            artifact_path = "{repo}/{path}/{name}".format(**artifact)
+
+        artifact_path = quote(artifact)
```
Comment (pcebul) [inline artifactory_cleanup/rules/base.py:208]: should be `artifact_path = quote(artifact_path)`

Comment (allburov) [reply artifactory_cleanup/rules/base.py]: You're so right, thank you for noticing such a bad mistake!

**Pull request merged**